src/components/viewer/GalleryViewer/index.js

```diff
@@ -24,72 +24,84 @@ import GalleryDocs from './GalleryDocs';
 
 import styles from './styles.scss';
 
-const propTypes = {
-    className: PropTypes.string,
-    url: PropTypes.string,
-    mimeType: PropTypes.string,
-    canShowIframe: PropTypes.bool,
-    showUrl: PropTypes.bool,
-    showScreenshot: PropTypes.bool,
-    onScreenshotCapture: PropTypes.func,
+function isUrlValid(url) {
+    return url && urlRegex.test(url);
+}
 
-    invalidUrlMessage: PropTypes.string,
-    cannotPreviewUrlMessage: PropTypes.string,
-};
+function Preview(props) {
+    const {
+        className,
 
-const defaultProps = {
-    className: '',
-    url: '',
-    mimeType: '',
-    canShowIframe: false,
-    showUrl: false,
-    showScreenshot: false,
-    onScreenshotCapture: undefined,
+        url,
+        mimeType,
 
-    invalidUrlMessage: undefined,
-    cannotPreviewUrlMessage: undefined,
-};
+        canShowIframe,
 
+        cannotPreviewUrlMessage,
+        invalidUrlMessage,
+        unsupportedTypeMessage,
 
-const isUrlValid = url => (url && urlRegex.test(url));
-
-const Preview = ({
-    className,
-    url,
-    mimeType,
-    canShowIframe,
-    previewError,
-    isHttps,
-    cannotPreviewUrlMessage,
-    invalidUrlMessage,
-    ...otherProps
-}) => {
-    if (isHttps && galleryMapping[mimeType] === galleryType.IMAGE) {
+        error,
+    } = props;
+
+    if (error) {
+        return (
+            <Message className={styles.errorUrl}>
+                {error}
+            </Message>
+        );
+    }
+
+    const isHttps = !!(url || '').match(/^https:\/\//) || window.location.protocol === 'http:';
+
+    if (galleryMapping[mimeType] === galleryType.IMAGE) {
+        // NOTE: Error can occur if
+        // 1. If there is no alternative https url and current url is http
+        const previewError = !isHttps;
+        if (previewError) {
+            return (
+                <Message className={styles.errorUrl}>
+                    {cannotPreviewUrlMessage || _ts('components.galleryViewer', 'cannotPreviewUrl')}
+                </Message>
+            );
+        }
         return (
             <GalleryImage
                 className={className}
                 imageUrl={url}
-                canShowIframe={canShowIframe}
-                {...otherProps}
             />
         );
     } else if (galleryMapping[mimeType] === galleryType.DOC) {
+        // NOTE: no need to check for https for GalleryDocs as it has google viewer as fallback
         return (
             <GalleryDocs
                 className={className}
                 docUrl={url}
                 mimeType={mimeType}
                 canShowIframe={canShowIframe}
                 notHttps={!isHttps}
-                {...otherProps}
```

Comment:
isn't this required?

Reply:
Nop, there is no remaining prop that galleryDocs accept.
I don't think we should let user override GalleryDocs props from GalleryViewer props anyway.